torch/_dynamo/variables/torch.py

```diff
@@ -220,77 +237,11 @@ def call_function(
             return TorchFunctionDisableVariable.create(tx)
 
 
-class TorchVariable(VariableTracker):
-    """Points to a module or method in torch.*"""
-
-    def __init__(self, value, **kwargs):
-        super().__init__(**kwargs)
-        if (
-            isinstance(value, collections.abc.Hashable)
-            and value in tensor_dunder_fns_remap
-        ):
-            value = tensor_dunder_fns_remap[value]
-
-        self.value = value
-
-        assert not isinstance(
-            value, (torch.dtype, torch.device)
-        ), "should use ConstantVariable"
-        # the remainder of this is just optional debug checks
-        try:
-            self_should_be_none = getattr(self.value, "__self__", None)
-        except RuntimeError as e:
-            assert "No such operator" in str(e), str(e)
-            self_should_be_none = None
-        except AssertionError as e:
-            assert "Unknown attribute" in str(e), str(e)
-            self_should_be_none = None
-
-        # assert "_ntuple.<locals>.parse" not in str(value)
-
-        if self_should_be_none is None:
-            pass
-        elif isinstance(self_should_be_none, types.ModuleType):
-            # weird ones like torch.nn.functional.avg_pool2d have __self__
-            name = self_should_be_none.__name__
-            assert re.match(r"^(torch|math)([.]|$)", name), f"__self__ set to {name}"
-        elif isinstance(
-            self_should_be_none, type(torch._C._get_tracing_state.__self__)
-        ):
-            # some _C functions have __self__ as a null capsule
-            pass
-        elif isinstance(self_should_be_none, torch_special_class_types):
-            pass
-        else:
-            raise AssertionError(f"{value} found with __self__ set")
+class TorchInGraphFunctionVariable(BaseTorchVariable):
+    """Points to a torch function/method that should be put in FX graph"""
 
     def __repr__(self):
-        return f"TorchVariable({self.value})"
-
-    def call_hasattr(self, tx, name):
-        result = hasattr(self.value, name)
-        return variables.ConstantVariable.create(result)
-
-    def reconstruct(self, codegen):
-        return torch_reconstruct(codegen, self.value)
-
-    def as_proxy(self):
-        return self.value
-
-    def python_type(self):
-        if isinstance(self.value, (torch.Tensor, torch.nn.Module, torch.device)):
-            return type(self.value)
-        if isinstance(self.value, type):
-            return type
```

Comment:
This special case was removed. Is this intentional? The special case does seem pretty suspicious.

Reply:
```TorchVariable``` still use the original code, but for ```BaseTorchVariable``` we are using the version w/o special cases, since the special cases were only wrapped as ```TorchVariable```.
And I think those special cases can be further cleaned up, ```torch.Tensor```, ```torch.nn.Module``` and ```torch.device``` are ```TensorVariable```, ```NNModuleVariable``` and ```ConstantVariable```. We should remove them from here.